process_context.go

```diff
@@ -0,0 +1,170 @@
+package scopeagent
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/opentracing/opentracing-go"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	environmentKeyPrefix = "CTX_"
+)
+
+var (
+	escapeMap          map[string]string
+	processSpanContext *opentracing.SpanContext
+)
+
+func init() {
+	// Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001
+	// consist solely of uppercase letters, digits, and the '_' (underscore)
+	escapeMap = map[string]string{
+		".": "__dt__",
+		"-": "__dh__",
+	}
+	if envCtx, err := ExtractFromEnvVars(os.Environ()); err == nil {
```

Comment:
I wouldn't do this here. I'd make this file be the context propagator for environment variables, and that's it. I'd then do the inject/extract somewhere else, so loading this file has no side effects.

Reply:
Ok, I will do it